static/js/settings.js

```diff
@@ -0,0 +1,64 @@
+(function() {
+	"use strict";
+
+    // TODO Find a way to make this reusable across the app?
+    function apiRequest(path, options) {
+        const cookieString = typeof document.cookie === "string" ? document.cookie : "";
+        const cookieStringArray = cookieString
+            .split(";")
+            .map(individualCookieString => individualCookieString.split("="))
+            .map(([cookieKey, cookieValue]) => [cookieKey.trim(), cookieValue.trim()]);
+        // Looks like the `argsIgnorePattern` option for ESLint doesn't like array destructuring:
+        // eslint-disable-next-line no-unused-vars
+        const [_csrfCookieKey, csrfCookieValue] = cookieStringArray.find(([cookieKey, _cookieValue]) => cookieKey === "csrftoken");
+        const headers = new Headers(options ? options.headers : undefined);
+        headers.set("X-CSRFToken", csrfCookieValue);
+        headers.set("Content-Type", "application/json");
+        headers.set("Accept", "application/json");
+        return fetch(
+            `/api/v1${path}`,
+            {
+                mode: "same-origin",
+                ...options,
+                headers: headers,
+            },
+        );
+    }
+
+    const settingsForm = document.getElementById("settings-form");
+    const labelCollectionCheckbox = document.getElementById("label-collection");
+    const labelCollectionOffWarning = document.getElementById("label-collection-off-warning");
+    const saveError = document.getElementById("save-error");
```

Comment:
nit: I think all of this is specific to the Settings page, but if not, I would recommend migrating from an ID to a `js-` prefix class for more usability.

Reply:
Whoops yeah, still have to get into that habit (haven't manually targeted elements from JS in a looong time): 753b47b

Feels a bit weird to be using class selectors for unique elements though :/